x-pack/plugin/esql/qa/server/multi-clusters/src/javaRestTest/java/org/elasticsearch/xpack/esql/ccq/MultiClusterSpecIT.java

```diff
@@ -0,0 +1,150 @@
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * This suite loads the data into either the local cluster or the remote cluster, then run spec tests with CCQ.
+ * TODO: Some spec tests prevents us from splitting data across multiple shards/indices/clusters
+ */
+public class MultiClusterSpecIT extends EsqlSpecTestCase {
+
+    static ElasticsearchCluster remoteCluster = Clusters.remoteCluster();
+    static ElasticsearchCluster localCluster = Clusters.localCluster(remoteCluster);
+
+    @ClassRule
+    public static TestRule clusterRule = RuleChain.outerRule(remoteCluster).around(localCluster);
+
+    public MultiClusterSpecIT(String fileName, String groupName, String testName, Integer lineNumber, CsvSpecReader.CsvTestCase testCase) {
+        super(fileName, groupName, testName, lineNumber, convertToRemoteIndices(testCase));
+    }
+
+    @Override
+    protected void shouldSkipTest(String testName) {
+        super.shouldSkipTest(testName);
+        assumeFalse("CCQ doesn't support enrich yet", hasEnrich(testCase.query));
+        assumeFalse("can't test with _index metadata", hasIndexMetadata(testCase.query));
+        assumeTrue("Test " + testName + " is skipped on " + Clusters.oldVersion(), isEnabled(testName, Clusters.oldVersion()));
+    }
+
+    @Override
+    protected String getTestRestCluster() {
+        return localCluster.getHttpAddresses();
+    }
+
+    @Override
+    protected RestClient buildClient(Settings settings, HttpHost[] localHosts) throws IOException {
+        RestClient localClient = super.buildClient(settings, localHosts);
+        HttpHost[] remoteHosts = parseClusterHosts(remoteCluster.getHttpAddresses()).toArray(HttpHost[]::new);
+        RestClient remoteClient = super.buildClient(settings, remoteHosts);
+        return twoClients(localClient, remoteClient);
+    }
+
+    static RestClient twoClients(RestClient localClient, RestClient remoteClient) throws IOException {
+        RestClient twoClients = mock(RestClient.class);
+        when(twoClients.performRequest(any())).then(invocation -> {
+            Request request = invocation.getArgument(0);
+            if (request.getEndpoint().contains("_query")) {
+                return localClient.performRequest(request);
+            } else if (request.getEndpoint().contains("_bulk")) {
+                if (randomBoolean()) {
+                    return remoteClient.performRequest(request);
+                } else {
+                    return localClient.performRequest(request);
+                }
+            } else {
+                localClient.performRequest(request);
+                return remoteClient.performRequest(request);
+            }
+        });
+        doAnswer(invocation -> {
+            IOUtils.close(localClient, remoteClient);
+            return null;
+        }).when(twoClients).close();
+        return twoClients;
```

Comment:
Cute.

Reply:
Probably worth javadoc.

Reply:
Yes, I added one in https://github.com/elastic/elasticsearch/pull/101640/commits/8c93a595f0d7801377db8611987ea6d2d3905af4